### DISCUSSION TITLE
Utility meter, non riesco a leggere il valore.

### DISCUSSION THREAD

**Discussion by Matty1184** [indomus/forum#3205] [HUB: Home Assistant]
CIao a tutti, 
non riesco a trovare il probelma.
ho creato un sensore che che mi somma l'energia delle varie fasce orarie, ma non riesco a ricavare il valore dallo stato.
i sensori sono quelli che chea in automatico "Utility meter", nel mio caso F1 F2 F3 in quanto parliamo di energia, nella consultazione stato dei tre sensori vedo il valore progressibo nello stato, l'ultimo periodo etc etc, se però lo inserisco all'interno della formula del value_template ho sempre zero.
questo è il sensore
``` 
sensor:
  - platform: template
    sensors:
      energia_prelevata_da rete_giornaliera_totale:
        friendly_name: "Energia Giornaliera Totale Prelevata da Rete"
        value_template: "{{ (state_attr('sensor.energia_prelevata_da_rete_giornaliera_f1', 'state')  | float(default=0)) + (state_attr('sensor.energia_prelevata_da_rete_giornaliera_f2', 'state')  | float(default=0)) + (state_attr('sensor.energia_prelevata_da_rete_giornaliera_f3', 'state')  | float(default=0)) }}"
        unit_of_measurement: "kWh" 
```
il problema è che 
```
{{ (state_attr('sensor.energia_prelevata_da_rete_giornaliera_f1', 'state')  | float(default=0)) }}
```
è sempre a zero, nonostante l'entità abbia un valore, tantoche se utilizzo la stringa in modelli
```
{{ states('sensor.energia_prelevata_da_rete_giornaliera_f1') | float(default=0) }}
```
restituisce il valore corretto.

dove sto sbagliando?

**Comment by KC-inDomus** [comment]
scusa, ma perché non ti fai un sensore globale, sempre tramite utility meter?

**Reply by Scialla** [reply]
A parte quanto ha detto KC, sbagli perché cerchi di usare lo stato come se fosse un attributo, cosa che non è.  Hai correttamente indicato la funzione `states()` che dovresti usare al posto di `state_attr()`

**Reply by Matty1184** [reply]
> scusa, ma perché non ti fai un sensore globale, sempre tramite utility meter?

Ciao KC, su questo non sapevo si potesse fare, visto che sono 3 fasce.
